What: Updated process for proposals to projects (#725)
cncf/tag-security: governance/process.md

for the suggestion, the work may not be done. Mentoring and co-leading are
available to anyone unsure or interested in becoming a project lead, often
suggestions have a limited scope of work and are ideal for individuals looking
to stretch their professional skills.

### Proposals

Proposals are ideas that are intended to become projects, where the submitter
intends to be the project lead. If a proposal is submitted that does not have a
project lead intent checked, it will be converted to a suggestion during triage.

#### Creating, discussing and accepting proposals

Each proposal is unique and might deviate slightly from the process below. For
example, a small addition may not require completion criteria. In general, we
encourage the process below to be followed to ensure that contributions are in
line with the [mission and charter](charter.md).

##### Proposal process

1. **Review existing issues:** [Look at the existing
   issues](https://github.com/cncf/tag-security/issues) and determine if any
   already exist. If an issue already exists that is marked as a suggestion but
   you are interested in leading, comment on the issue and reach out in the

>    [#tag-security-triage](https://cloud-native.slack.com/archives/CJX8EBF1D)

Author: I think it depends.  the general channel (to me) seems like the go-to place for help or posting things.  governance does as well.  triage has a fair amount of traffic but is really the right place for that notice - or governance?

>    channel to bring attention to it. Skip to the 3rd step.

Author: ```suggestion
   channel to bring attention to it. Skip to ask for collaboration.
```

1. **Raise an Issue:** [Create an
   issue](https://github.com/cncf/tag-security/issues/new) that outlines the
   problem to be solved. Use the proposal template if you are interested in
   leading the work and _be sure to complete the project lead intent checkbox_,
   or use the suggestion template if you would like someone else to lead the
   effort. Try to complete the template as much as possible.

1. **Ask the group for collaboration:** Rather than immediately beginning work

>    on a solution, bring the issue up for discussion in a Security TAG meeting.

Reviewer: I tried to make it clear below that discussion via slack/email/etc is welcome.  The more explicit this is, though, the harder it is to expect people to read the whole thing, so we should also balance concise instructions.

   The following guidance shows common steps, though communication often happens
   in different sequence. The key outcome is that there is opportunity for input
   across different channels, with thoughtfulness to accessibility across
   timezone and communication medium. We also encourage outreach outside of the
   group, when there are experts who might share insights (via invited
   presentation) or wish to get involved.

   A) On slack, share the issue link and ask whether others are interested in

>    the problem, if so, let them know to join the planned meeting (see below)

Reviewer: again, let's keep meetings optional, can add "if they can" but the planned meeting should be noted on the issue and we should steer people to always update the issue -- things work well when everyone looks to the info in the issue as the source of truth, so it's friendly to new participation and leadership team or other project leaders and other groups can also look to the issue to see what is going on

>    the problem, if so, let them know to join the planned meeting (see below)

Author: ```suggestion
   the problem, if so, refer them to the issue which has content on discussions/meetings
```

   where the issue will be introduced or where to stay tuned for more
   information. provide the issue link and encourage any feedback on your
   proposed solution or activity.

>    B) Choose an upcoming meeting in **_either region_** where you or another

Reviewer: ```suggestion
   B) Choose an upcoming meeting where you or another
```
let's not encode two regions!    We may someday split Europe/Africa from Americas

    and on-going stream of work -- the latter would typically be proposed only
    after at least one, usually multiple, projects have been completed.
    * At least one person is recommended or nominated as a potential lead.
      of done", ideally with validation by the target audience. Also note in the
      issue if it will be a time-bounded project with a defined deliverable or
      and on-going stream of work -- the latter would typically be proposed only
      after at least one, usually multiple, projects have been completed.
    * At least one person is recommended or nominated as a potential lead --
      usually the author of the proposal.
    * Those interested in working on the solution comment on the issue so
    coordination may begin and set up time or expectations with others to
    begin work.
    * A Chair or Technical Lead proposes this as a roadmap proposal or
    agrees to act as a "sponsor."
      * Sponsor: takes responsibility to ensure that progress
      is tracked and that outcomes are reported to the group, including
      proposing to close the issue if there is not sufficient activity to

>       sustain the effort.

Reviewer: I dunno -- if no one is interested in sponsoring it, which typically happens if there's too much going on right now, do we want to encourage scoping/investment in getting the thing done?  In practice, I've rarely seen something well scoped without the collaboration of a Chair or TL

Author: IME - scoping is driven my the group or author with feedback from the Chair TL.  Doesn't mean it doesn't happen, and ultimately those conversations that drive scoping are usually with the chair/TL that is willing to sponsor (as sponsor they "sign off" on the scope as well?).  I've not seen where we don't sponsor due to timing, rather we sponsor but postpone the work.

Reviewer: There are currently 21 [open proposals](https://github.com/cncf/tag-security/issues?q=is%3Aissue+is%3Aopen+label%3Aproposal) and I thought most were open due to lack of bandwidth. (If TL/Chairs weren't already working on so many things in parallel, then I figured some of those would be moving forward.)

Author: I think a lot of these no longer have interest.  its worth the time to sort through them and triage them accordingly.  thanks for bringing this up.

>    B) **Close**: Closure may occur as a result of lack of interest (stale for 90

Reviewer: When originally written, this option was considered way earlier in the process.  At any time, the proposal may be closed by leadership team as out of scope of the group, but we agreed not to do that before it was discussed at a meeting (hence original order) but it we put scoping first then this doesn't feel right here.

Author: A lot of suggestions or proposals that come to the group arent fully formed.  Scoping allows us to be more definitive in what is or is not aligned with the charter.

Reviewer: yes, agreed for something that is clearly within our charter.  If something comes to us that is not appropriate to the group, then we want to close early in the process.

   it is beneficial to present the work to the broader group to gather a final
   review, solicit additional feedback, or seek recommendations. These should be
   placed in the planned meetings section with a project member as the point of
   contact to present.

1. **Pull Requests.** Completed work should result in a Pull Request (PR). At
   minimum, an update to one of the group documents or roadmap indicating that
   the work was done. Typically projects will result in an artifact that will
   contribute to the information in this repository.

1. **Discuss the work at a meeting.** If an objection to a PR is made either in
   a comment of the PR or during a meeting, the person making the objection and
   the person making the proposal will be given time to present their view at

>    the next meeting. If there are not objections, or if all concerns have been

Author: ```suggestion
   the next meeting. If there are no objections, or if all concerns have been
```

   addressed, and the Pull Request has been stable for 24 hours, a Chair will
   add it to the agenda for an upcoming meeting. Ideally, members who
   contributed to the project will attend that meeting to present their work or
   answer questions.

1. **Vote, if required.** In some cases, there's consensus to accept a PR, and a

>    vote is not required. If there’s not consensus among the group, a formal vote

Reviewer 2: Nit: `not` -> `no`

>    vote is not required. If there’s not consensus among the group, a formal vote

Author: ```suggestion
   vote is not required. If there’s no consensus among the group, a formal vote
```

   is required. A comment will be left on the PR prompting members to vote and
   indicating the time the vote will close. Only one member from each company
   should vote. Members will vote by leaving a comment in the Pull Request to
   indicate their vote for or against. Members will have a week after the vote
   begins to leave their vote. Quorum is taken to be 2/3 the number of companies
   who have been active in the past month (via issue comment or meeting

>    attendance as recorded in the public minutes).

Reviewer 3: I'm a little unsure about the voting system, I think we haven't had a situation where this is needed yet, and wondering if it would be best to hold off on this until we see the need to... I think leaving it to the decision of leadership (or a unanimous decision among leadership) would be good enough.

Reviewer 3: If we do decide to go with voting, I think that we need to have a bit more "system" around this. Voting needs to ensure visibility of the vote happening, maybe via email list. There needs to be proof of participation provided with the vote so we don't have to go around digging up history. We need to ensure that there isn't bad faith with editing meeting notes to include past participation since there is now an incentive to do so.

Reviewer 2: I agree with Brandon that the final decision should be left with the leadership team. We could do a majority or 2/3rd vote in the leadership team. We can encourage community members to submit "non-binding" votes to show support or lack of it.

1. **Raise an Issue:** [Create an
   issue](https://github.com/cncf/tag-security/issues/new) that outlines the
   problem to be solved. Use the proposal template if you are interested in
   leading the work and _be sure to complete the project lead intent checkbox_,
   or use the suggestion template if you would like someone else to lead the
   effort. Try to complete the template as much as possible.

1. **Ask the group for collaboration:** Rather than immediately beginning work
   on a solution, bring the issue up for discussion. The following guidance
   shows common steps, though communication often happens in different sequence.
   The key outcome is that there is opportunity for input across different
   channels, with thoughtfulness to accessibility across timezone and
   communication medium. We also encourage outreach outside of the group, when
   there are experts who might share insights (via invited presentation) or wish
   to get involved.

   A) On slack, share the issue link and ask whether others are interested in
   the problem, if so, refer them to the issue which has content on
   discussions/meetings where the issue will be introduced or where to stay

>    tuned for more information. provide the issue link and encourage any feedback

Reviewer 2: Nit: `provide` -> `Provide`

   on your proposed solution or activity.

   B) Choose an upcoming meeting where you or another group member who is
   interested in working on the project is able to attend, then add the issue to
   the meeting agenda: include a link and the name of the person who will
   present the proposal in the "Planned Meeting" area of the [meeting
   notes][https://github.com/cncf/tag-security#meeting-times]. Then at the
   meeting:

   * The presenter should screen share the github issue (or ask the meeting
     facilitator ahead of time to do so) and explain the motivation, expected
     outcome, ideas that they have for how it might happen, and ask if others
     have ideas or questions.
   * After a short discussion, people should be invited to chime in on the

>      github issue and also mention of they are interested in collaborating. This

Reviewer 2: Nit: `of` -> `if`
